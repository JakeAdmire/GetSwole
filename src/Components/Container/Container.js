import React, { Component } from 'react'
import { connect } from 'react-redux';
import { Text, View } from 'react-native';
import { Card } from 'react-native-elements';
// 
import { fetchRoutines } from '../../Thunks/fetchRoutines';

export class Container extends Component  {

  displayCards = () => {
    const { routines } = this.props;
    return routines.data && routines.data.length
      ? routines.data.map(routine => (
        <View key={routine.id}>
          <Card>
            <Text>{routine.attributes.name}</Text>
            {
              routine.attributes.exercises.map(exercise => (
                <Text key={exercise.id}>{exercise.name}</Text>
              ))
            }
          </Card>
        </View>
      ))
      : <Text>No routines scheduled for this day</Text>
  }

  render () {
    const { date, routines, loading } = this.props;

    return loading
      ? <Text>loading..</Text>
      : <View>{ this.displayCards() }</View>
  }
}

export const mapStateToProps = (state) => ({
  date: state.date,
  routines: state.routines,
  loading: state.loading
})

export default connect(mapStateToProps, null)(Container);
