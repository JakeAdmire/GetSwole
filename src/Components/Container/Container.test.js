import React from 'react';
import { shallow } from 'enzyme';
// 
import { Container, mapStateToProps } from './Container';

let wrapper;

let mockDate = '2019 5 25';
let semanticDateMock = '25th March, 2019';
let loadingMock = false;
let navToolMock = { navigate: jest.fn() };
let routinesMock = {
  data: [
    {
      id: "4",
      type: "routine",
      attributes: {
        name: "Arm Day",
        exercises: [
          {
            id: 50,
            name: "Pushups",
            category: "Strength",
            equipment_required: "Body Only",
            muscle: "Chest",
            reps: 10,
            sets: 4,
            duration: null,
            weight: null
          },
          {
            id: 70,
            name: "Chin-Up",
            category: "Strength",
            equipment_required: "Body Only",
            muscle: "Lats",
            reps: 5,
            sets: 4,
            duration: null,
            weight: null
          },
          {
            id: 104,
            name: "Barbell Curl",
            category: "Strength",
            equipment_required: "Barbell",
            muscle: "Biceps",
            reps: 10,
            sets: 4,
            duration: null,
            weight: null
          }
        ]
      }
    }
  ]
};

let routinesMock2 = {
  data: [
    {
      id: "4",
      type: "routine",
      attributes: {
        name: "Arm Day",
        exercises: [
          {
            id: 50,
            name: "Pushups",
            category: "Strength",
            equipment_required: "Body Only",
            muscle: "Chest",
            reps: 10,
            sets: 4,
            duration: null,
            weight: null
          }
        ]
      }
    },
    {
      id: "5",
      type: "routine",
      attributes: {
        name: "Leg Day",
        exercises: [
          {
            id: 75,
            name: "Squats",
            category: "Strength",
            equipment_required: "Body Only",
            muscle: "Legs",
            reps: 10,
            sets: 4,
            duration: null,
            weight: null
          }
        ]
      }
    }
  ]
};

describe('Container', () => {

  beforeEach(() => {
    wrapper = shallow(
      <Container  date={mockDate}
                  semanticDate={semanticDateMock}
                  loading={loadingMock}
                  navTool={navToolMock}
                  routines={routinesMock} />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  })

  describe('handleAddNewRoutine()', () => {

    it.skip('should be invoked on button press', () => {
      let mockSpy = jest.spyOn(wrapper.instance(), 'handleAddNewRoutine');
      const addRoutineButton = wrapper
        .findWhere(node => node.prop('testID') === 'add-routine-button');

      addRoutineButton.props().onPress(); 

      expect(mockSpy).toHaveBeenCalled();
    })

    it('should invoke navTool.navigate with a string', () => {
      wrapper.instance().handleAddNewRoutine();
      expect(navToolMock.navigate).toHaveBeenCalledWith("routinePage");
    })

  })

  describe('displayCards()', () => {

    it('should match snapshot', () => {
      let mockSpy = jest.spyOn(wrapper.instance(), 'displayCards');
      expect(mockSpy).toMatchSnapshot();

      wrapper.setProps({ routines: routinesMock2 });
      expect(mockSpy).toMatchSnapshot();

      wrapper.setProps({ routines: {} });
      expect(mockSpy).toMatchSnapshot();
    })

    it('should call renderButton', () => {
      let mockSpy = jest.spyOn(wrapper.instance(), 'renderButton');

      wrapper.instance().displayCards();

      expect(mockSpy).toHaveBeenCalled();
    })

  })

  describe('renderButton()', () => {

    it('should match snapshot', () => {
      let mockSpy = jest.spyOn(wrapper.instance(), 'renderButton');

      expect(mockSpy).toMatchSnapshot();
    })

  })

})

describe('mapStateToProps', () => {

  it('should return a props object', () => {
    let mockState = {
      date: mockDate,
      semanticDate: semanticDateMock,
      routines: routinesMock,
      loading: loadingMock
    };
    let results = mapStateToProps(mockState);

    expect(results).toEqual(mockState)
  })

})
