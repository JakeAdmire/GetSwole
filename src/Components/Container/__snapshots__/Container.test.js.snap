// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container displayCards() should match snapshot 1`] = `[MockFunction]`;

exports[`Container displayCards() should match snapshot 2`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2D71A8",
            "color": "#ACC6D0",
            "height": "100%",
            "padding": 20,
            "width": 750,
          }
        }
      >
        <Component
          style={
            Object {
              "width": 730,
            }
          }
        >
          <RalewayText
            style={
              Object {
                "color": "#53CFFF",
                "fontSize": 16,
              }
            }
            text="Here's the schedule for"
          />
          <RalewayBoldText
            style={
              Object {
                "color": "#FFF",
                "fontSize": 16,
                "marginBottom": 10,
              }
            }
            text="25th March, 2019"
          />
        </Component>
        <Themed.ListItem
          bottomDivider={true}
          containerStyle={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "width": 730,
            }
          }
          leftIcon={
            <Themed.Icon
              color="#2D71A8"
              name="trash"
              onPress={[Function]}
              type="font-awesome"
            />
          }
          rightIcon={
            <Themed.Icon
              color="#2D71A8"
              name="chevron-circle-right"
              type="font-awesome"
            />
          }
          subtitle="1 exercises"
          subtitleStyle={
            Object {
              "color": "#53CFFF",
              "fontFamily": "raleway",
            }
          }
          title="Arm Day"
          titleStyle={
            Object {
              "color": "#2D71A8",
              "fontFamily": "raleway",
            }
          }
        />
        <Themed.ListItem
          bottomDivider={true}
          containerStyle={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "marginBottom": 10,
              "width": 730,
            }
          }
          leftIcon={
            <Themed.Icon
              color="#2D71A8"
              name="trash"
              onPress={[Function]}
              type="font-awesome"
            />
          }
          rightIcon={
            <Themed.Icon
              color="#2D71A8"
              name="chevron-circle-right"
              type="font-awesome"
            />
          }
          subtitle="1 exercises"
          subtitleStyle={
            Object {
              "color": "#53CFFF",
              "fontFamily": "raleway",
            }
          }
          title="Leg Day"
          titleStyle={
            Object {
              "color": "#2D71A8",
              "fontFamily": "raleway",
            }
          }
        />
        <Component
          style={
            Object {
              "width": 730,
            }
          }
        >
          <RalewayText
            style={
              Object {
                "color": "#53CFFF",
                "fontSize": 16,
                "marginBottom": 10,
                "marginTop": 30,
              }
            }
            text="Still not tired? Add another routine:"
          />
        </Component>
        <Component
          style={
            Object {
              "width": 730,
            }
          }
        >
          <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "#FFF",
                "borderRadius": 10,
              }
            }
            icon={
              <Themed.Icon
                color="#2D71A8"
                iconStyle={
                  Object {
                    "position": "absolute",
                    "right": -75,
                  }
                }
                name="plus"
                size={18}
                type="font-awesome"
              />
            }
            iconRight={true}
            onPress={[Function]}
            testID="add-routine-button"
            title="Add New Routine"
            titleStyle={
              Object {
                "color": "#2D71A8",
                "fontFamily": "raleway-bold",
              }
            }
          />
        </Component>
      </Component>,
    },
  ],
}
`;

exports[`Container displayCards() should match snapshot 3`] = `
[MockFunction] {
  "calls": Array [
    Array [],
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2D71A8",
            "color": "#ACC6D0",
            "height": "100%",
            "padding": 20,
            "width": 750,
          }
        }
      >
        <Component
          style={
            Object {
              "width": 730,
            }
          }
        >
          <RalewayText
            style={
              Object {
                "color": "#53CFFF",
                "fontSize": 16,
              }
            }
            text="Here's the schedule for"
          />
          <RalewayBoldText
            style={
              Object {
                "color": "#FFF",
                "fontSize": 16,
                "marginBottom": 10,
              }
            }
            text="25th March, 2019"
          />
        </Component>
        <Themed.ListItem
          bottomDivider={true}
          containerStyle={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "width": 730,
            }
          }
          leftIcon={
            <Themed.Icon
              color="#2D71A8"
              name="trash"
              onPress={[Function]}
              type="font-awesome"
            />
          }
          rightIcon={
            <Themed.Icon
              color="#2D71A8"
              name="chevron-circle-right"
              type="font-awesome"
            />
          }
          subtitle="1 exercises"
          subtitleStyle={
            Object {
              "color": "#53CFFF",
              "fontFamily": "raleway",
            }
          }
          title="Arm Day"
          titleStyle={
            Object {
              "color": "#2D71A8",
              "fontFamily": "raleway",
            }
          }
        />
        <Themed.ListItem
          bottomDivider={true}
          containerStyle={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 10,
              "marginBottom": 10,
              "width": 730,
            }
          }
          leftIcon={
            <Themed.Icon
              color="#2D71A8"
              name="trash"
              onPress={[Function]}
              type="font-awesome"
            />
          }
          rightIcon={
            <Themed.Icon
              color="#2D71A8"
              name="chevron-circle-right"
              type="font-awesome"
            />
          }
          subtitle="1 exercises"
          subtitleStyle={
            Object {
              "color": "#53CFFF",
              "fontFamily": "raleway",
            }
          }
          title="Leg Day"
          titleStyle={
            Object {
              "color": "#2D71A8",
              "fontFamily": "raleway",
            }
          }
        />
        <Component
          style={
            Object {
              "width": 730,
            }
          }
        >
          <RalewayText
            style={
              Object {
                "color": "#53CFFF",
                "fontSize": 16,
                "marginBottom": 10,
                "marginTop": 30,
              }
            }
            text="Still not tired? Add another routine:"
          />
        </Component>
        <Component
          style={
            Object {
              "width": 730,
            }
          }
        >
          <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "#FFF",
                "borderRadius": 10,
              }
            }
            icon={
              <Themed.Icon
                color="#2D71A8"
                iconStyle={
                  Object {
                    "position": "absolute",
                    "right": -75,
                  }
                }
                name="plus"
                size={18}
                type="font-awesome"
              />
            }
            iconRight={true}
            onPress={[Function]}
            testID="add-routine-button"
            title="Add New Routine"
            titleStyle={
              Object {
                "color": "#2D71A8",
                "fontFamily": "raleway-bold",
              }
            }
          />
        </Component>
      </Component>,
    },
    Object {
      "type": "return",
      "value": <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2D71A8",
            "color": "#ACC6D0",
            "height": "100%",
            "padding": 20,
            "width": 750,
          }
        }
      >
        <Component
          style={
            Object {
              "width": 730,
            }
          }
        >
          <RalewayText
            style={
              Object {
                "color": "#53CFFF",
                "fontSize": 16,
              }
            }
            text="You don't yet have anything scheduled for"
          />
          <RalewayBoldText
            style={
              Object {
                "color": "#FFF",
                "fontSize": 16,
                "marginBottom": 10,
              }
            }
            text="25th March, 2019"
          />
        </Component>
        <Component
          style={
            Object {
              "width": 730,
            }
          }
        >
          <RalewayText
            style={
              Object {
                "color": "#53CFFF",
                "fontSize": 16,
                "marginBottom": 10,
                "marginTop": 30,
              }
            }
            text="Let's Fix That:"
          />
        </Component>
        <Component
          style={
            Object {
              "width": 730,
            }
          }
        >
          <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "#FFF",
                "borderRadius": 10,
              }
            }
            icon={
              <Themed.Icon
                color="#2D71A8"
                iconStyle={
                  Object {
                    "position": "absolute",
                    "right": -75,
                  }
                }
                name="plus"
                size={18}
                type="font-awesome"
              />
            }
            iconRight={true}
            onPress={[Function]}
            testID="add-routine-button"
            title="Add New Routine"
            titleStyle={
              Object {
                "color": "#2D71A8",
                "fontFamily": "raleway-bold",
              }
            }
          />
        </Component>
      </Component>,
    },
  ],
}
`;

exports[`Container renderButton() should match snapshot 1`] = `[MockFunction]`;

exports[`Container should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Container should match snapshot 2`] = `ShallowWrapper {}`;
